lib/shared/bucketing/src/segmentation.ts

```diff
@@ -57,6 +63,32 @@ type FilterFunctionsBySubtype = {
     [key in UserSubType]: (data: any, filter: AudienceFilterOrOperator) => boolean
 }
 
+function filterForAudienceMatch({
+    operator, data, featureId, isOptInEnabled, audiences = {}
+}: {
+    operator: AudienceFilterOrOperator, data: any, featureId: string, isOptInEnabled: boolean
+    audiences?:{ [id: string]: Omit<PublicAudience<string>, '_id'> }
+}): boolean {
+    const comparator = operator.comparator
+    let userInAudience = false
+    // Recursively evaluate every audience in the _audiences array
+    operator._audiences?.forEach((_audience: string) => {
```

Comment:
might look cleaner if you use a proper `for` loop instead of the for each

Reply:
yea I was thinking the same thing. I was too excited to get back to es6 stuff hehe